### PR TITLE
Reduce dependencies on RecoEgamma and requirements for RecoEgamma

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

To make cmssw a bit more modular, I looked at the RecoEgamma packages and tried to remove all unused requirements. At the same time, I looked at the spots where other parts of cmssw depend on RecoEgamma and tried to remove these dependencies where possible. The ElectronIdentification and PhotonIdentification packages got special attention, and all sources in these libraries which are only used for plugins were moved into the plugins directory to make this more explicit.

#### PR validation:

CMSSW compiles and local matrix tests pass.

#### if this PR is a backport please specify the original PR:

No backport intended.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Review by perrotta — COMMENTED** [review]
Nice cleaning.
Maybe you could further adjust the header guards in those header files that you moved somewhere else, where needed.

**Comment by guitargeek** [comment]
Hi @perrotta, thanks for you comments! I implemented them both in the last commit and additionally renamed the now moved `FTSFromVertexToPointFactory.h` because it was not really a factory. just a free function.

File: TrackingTools/TrajectoryState/interface/ftsFromVertexToPoint.h [hunk 1]
```diff
@@ -1,9 +1,7 @@
...
-#define FTSFROMVERTEXTOPOINTFACTORY_H
+#ifndef TrackingTools_TrajectoryState_ftsFromVertexToPoint_h
+#define TrackingTools_TrajectoryState_ftsFromVertexToPoint_h
 // -*- C++ -*-
 //
-// Package:    EgammaElectronAlgos
-// Class:      FTSFromVertexToPointFactory
 //
 /**
  *  Generates a FreeTrajectoryState from a given measured point, vertex
```
Comment (perrotta) [inline TrackingTools/TrajectoryState/interface/ftsFromVertexToPoint.h]: The description should be updated to reflect your last change.
Also add your name as "Modified by" after L16

Comment (guitargeek) [reply TrackingTools/TrajectoryState/interface/ftsFromVertexToPoint.h]: Which part exactly should I update, you want me to put this package tag back in with the new location? Or is it fine leaving it out and I should just update the description text with the example? Thanks

Comment (perrotta) [reply TrackingTools/TrajectoryState/interface/ftsFromVertexToPoint.h]: I was thinking just at the example, when it says
```
 *  FTSFromVertexToPointFactory myFTS; myFTS(xmeas, xvert, momentum, charge);
 *  gives a FreeTrajectoryState of a track which comes from xvert to xmeas. 
```
Everything else seems still up to date to me

**Comment by perrotta** [comment]
please test

**Comment by jfernan2** [comment]
+dqm

**Comment by perrotta** [comment]
+1
- Code cleaning applied as outlined in the PR description
- As such, the changes are rather technical: no effect is visible on jenkins tests outputs, nor any difference in timing performance was measured in the 10224 (TTbar with PU) workflow

**Comment by santocch** [comment]
+1

**Comment by Martin-Grunewald** [comment]
+1

**Comment by tlampen** [comment]
+1

**Comment by silviodonato** [comment]
@benkrikler  @rekovic @chayanit  @pgunnell  @zhenhu please review/sign

**Comment by guitargeek** [comment]
Hi, unfortunately I had to rebase to fix a conflict in `DQMOffline/Trigger/plugins/LepHTMonitor.cc`.

**Comment by perrotta** [comment]
please test

**Comment by jfernan2** [comment]
+1
No dqm changes

**Comment by perrotta** [comment]
+1
- Only rebased since last reco signature in https://github.com/cms-sw/cmssw/pull/28684#issuecomment-570802837

**Comment by tocheng** [comment]
+1

**Comment by chayanit** [comment]
+1

**Comment by santocch** [comment]
+1

**Comment by perrotta** [comment]
Just to avoid having to rebase once more:
- @benkrikler,@rekovic could you please have a look and sign, if you think so, for L1?
- @Martin-Grunewald you already signed for HLT before the rebase, I think you can redo it now

**Comment by Martin-Grunewald** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Comment by silviodonato** [comment]
merge
@rekovic this PR changes only one line of L1

**Pull request merged**